source/NPC.cpp

```diff
@@ -705,6 +859,14 @@ NPC NPC::Instantiate(const PlayerInfo &player, map<string, string> &subs, const
 		if(result.personality.IsDerelict())
 			ship->Disable();
 
+		// Use the destinations stored in the instantiated NPC, in case they were auto-generated.
```

Comment:
I read this comment and think "do the correct thing, don't do the incorrect thing."
"Auto-generated" isn't even a good description. At best, that could describe the case where the destination from the mission is used, that would be automatic, since the user doesn't do anything but ask for whatever the mission is using to be used here. Locations selected from a filter are not auto-generated; a person defined that filter.
```suggestion
```